File: bin/src/EjerciciosDeRepaso.py
import math
"""se pone al principio por su hay más de uno que lo usa"""
print("inicio ejercicio 1")
print("dame un número")
numuno=10
numdos=10.5
producto=numuno*numdos
"""dos formas de ponerlo, con la , ya lo cambia a string (str) y deja un espacio y con el + hay que convertirlo a string con str ya que no interpreta que es una cadena por lo que no deja el espacio"""
print("la multiplicacion de los dos números es:", (producto))
print("la multiplicación de los dos números es:"+ str(producto))
print("   fin ejercicio 1")

print("inicio ejercicio 2")

pi=3.1416
radio=10
print("el área del círulo es:",(pi*radio*radio))
"""otra forma"""
area=math.pi*math.pow(radio,2)
area=math.pi*(radio**2)
area=math.pi*(radio^2)
area=math.pi*(radio*radio)



print("   fin ejercicio 2")

print("inicio ejercicio 3")
a=5
b=10
c=15
discriminante=b**2 -4*a*c
if discriminante>0:
    """si no importamos math no funciona la formula"""    
    raizdiscriminante= math.sqrt(discriminante)
    x1 = (-b + raizdiscriminante) / (2 * a) 
    x2 = (-b - raizdiscriminante) / (2 * a)
    print("las soluciones son x1=",x1," y x2=",x2)
elif discriminante==0:
    x = -b / (2 * a)
    print("la solución es x=",x)
else:
    print("No hay solución real debido a que el discriminante es negativo")
print("   fin ejercicio 3")

print("inicio ejercicio 4")
nombre="David"
edad=50
print("Tu nombre es",nombre,"y tu edad",edad,"años")
print("Tu nombre es "+nombre+" y tu edad "+str(edad)+" años")
print("   fin ejercicio 4")
print("inicio ejercicio 5")
"""como me quedo con la parte entera de una division en pythonn
Para obtener la parte entera de una división en Python, puedes utilizar el operador de división entera //. Este operador realiza una división y devuelve el cociente entero, descartando cualquier parte fraccionaria. """
dinero=237.88
centimos=dinero*100
moneda=centimos/200
print("monedas de dos euros: " + str (moneda))
moneda=centimos//200
print("monedas de dos euros: " + str (moneda))
moneda=int(centimos/200)
print("monedas de dos euros: " + str (moneda))
"""es lo mismo al ponerle , ya lo pasa a str y le da un espacio"""
print("monedas de dos euros:" ,moneda)

centimos=centimos-(moneda*200)
moneda=int(centimos/100)
print("monedas de 1 Euro: "+str (moneda))
centimos=centimos-(moneda*100)
moneda=int(centimos/50)
print("monedas de 50 cts "+ str(moneda))
centimos=centimos-(moneda*50)
moneda=int(centimos/20)
print("monedas de 20 cts: "+ str(moneda))
centimos=centimos-(moneda*20)
moneda=int(centimos/10)
print("monedas de 10 cts: "+ str(moneda))
centimos=centimos-(moneda*10)
moneda=int(centimos/5)
print("monedas de 5 cts: "+str(moneda))
centimos=centimos-(moneda*5)
moneda=int(centimos/2)
print("monedas de 2 cts: "+str(moneda))
centimos=centimos-(moneda*2)
print("monedas de 1 cts: "+ str(moneda))
print("   fin ejercicio 5")

print("CONDICIONALES inicio ejercicio 1")
numtres=5
if numtres%2==0:
    print("el número es par")
else:
    print("el número es impar")
print("CONDICIONALES    fin ejercicio 1")

print("CONDICIONALES inicio ejercicio 2")
edaddos=19
if edaddos>=19:
    print("tienes carnet de conducir?")
"""si tiene menos de 18 no le preguntamos nada"""


print("CONDICIONALES    fin ejercicio 2")

print("CONDICIONALES inicio ejercicio 3")
aa=3
bb=5
cc=1
if aa<=bb and aa<=cc:
    menor=aa
elif bb<=aa and bb<=cc:
    menor=bb
else:
    menor=cc
print("el número menor es : "+ str(menor))

print("CONDICIONALES    fin ejercicio 3")

print("CONDICIONALES inicio ejercicio 4")
num1=10
num2=1
num3=5
if num1>=num2 and num1>=num3:
    mas=num1
    if num2>=num3:
        menos=num3
    else:
        menos=num2
elif num2>=num1 and num2>=num3:
    mas=num2
    if num1>=num3:
        menos=num3
    else:
        menos=num1
else:
    mas=num3
    if num2>=num3:
        menos=num3
    else:
        menos=num2
print("el número menor es: "+str(menos)+" y el mayor es :"+str(mas))

print("CONDICIONALES    fin ejercicio 4")

print("CONDICIONALES inicio ejercicio 5")
real=100
otrocaso=0
if real<50:
    cuadrado=pow(real,2)
    print("su cuadrado es "+str(cuadrado))
    otrocaso=otrocaso+1
if real==25 or real==30 or real==75:
    raiz=pow(real,1/3)
    print("la raiz cúbica es : "+str(raiz))
    otrocaso+=1
if real==10 or real>100 or (real>=3 and real<=8) or (real>=77 and real<=90):
    division=real/10
    print("la división entre 10 es : "+str(division))
    otrocaso+=1
if otrocaso==0:
    print("el número introducido es : "+ str(real))


print("CONDICIONALES    fin ejercicio 5")

print("CONDICIONALES inicio ejercicio ")
print("CONDICIONALES    fin ejercicio ")

File: src/src/EjerciciosDeRepaso.java
package src;

public class EjerciciosDeRepaso {
    public static void main(String[] args) throws Exception {

        System.out.println("inicio ejercicio 1");
        /*
         * 1 pedir al usuario dos números, uno entero y otro real y mostrar su producto.
         * al no saber como recoger el número solo se pide
         */
        System.out.println("dame un número");
        int numuno = 10;
        System.out.println("dame un número con decimales");
        double numdos = 10.5;
        double producto = numuno * numdos;
        System.out.println("la multiplicación de los dos números es : " + producto);

        System.out.println("   fin ejercicio 1");

        System.out.println("inicio ejercicio 2");
        /* 2 Calcular el área de un círculo cuyo radio se le preguntará al usuario. */
        double pi = 3.1416;
        int radio = 10;
        double area = pi * (radio*radio);
        System.out.println("el área del círculo es:" + area);

        /*otra forma */
        /*double area =Math.PI*(radio^2);
        System.out.println(area);*/

        System.out.println("   fin ejercicio 2");

        System.out.println("inicio ejercicio 3");
        /*
         * 3 Pedir al usuario los coeficientes (a, b, c) de un polinomio de segundo
         * grado y mostrar las dos raíces.
         */
        int a = 5, b = 10, c = 15;
        double discriminante = Math.pow(b, 2) - 4 * a * c;
        /*
         * si es positivo hay dos resultados si es 0 hay uno y si es negativo no hay
         * solución
         */
        if (discriminante > 0) {
            double raizdiscriminante = Math.sqrt(discriminante);
            double x1 = (-b + raizdiscriminante) / (2 * a);
            double x2 = (-b - raizdiscriminante) / (2 * a);
            System.out.println("las soluciones son x1=" + x1 + " y x2=" + x2);
        } else if (discriminante == 0) {
            double x = -b / (2 * a);
            System.out.println("la solución es x=" + x);

        } else {
            System.out.println("No hay solución real debido a que el discriminante es negativo");
        }
        System.out.println("   fin ejercicio 3");

        System.out.println("inicio ejercicio 4");
        /*
         * 4 Preguntar al usuario su nombre (texto) y su edad (valor entero) y mostrar
         * por pantalla un texto del estilo del siguiente: Te llamas Lucía y tienes 19
         * años.
         */

        String nombre = "David";
        int edad = 50;
        System.out.println("Te llamas " + nombre + " y tienes " + edad + " años");

        System.out.println("   fin ejercicio 4");

        System.out.println("inicio ejercicio 5");
        /*
         * 5 Preguntar una cantidad en euros e indicar la cantidad de monedas de cada
         * tipo que se necesitarán para entregar esa cantidad. Se utilizarán las monedas
         * de mayor valor posible.
         */

        double dinero = 237.88;
        /*el (int) (dinero*100) es parsearlo */
        int centimos = (int) (dinero * 100);
        int moneda =  (centimos / 200);
        System.out.println("monedas de dos Euros: " + moneda);
        centimos = centimos - (moneda * 200);
        moneda = (centimos / 100);
        System.err.println("monedas de 1 Euro: " + moneda);
        centimos = centimos - (moneda * 100);
        moneda = (centimos / 50);
        System.out.println("monedas de 50 cts: " + moneda);
        centimos = centimos - (moneda * 50);
        moneda = (centimos / 20);
        System.out.println("monedas de 20 cts " + moneda);
        centimos = centimos - (moneda * 20);
        moneda = (centimos / 10);
        System.out.println("monedas de 10 cts " + moneda);
        centimos = centimos - (moneda * 10);
        moneda = (centimos / 5);
        System.out.println("monedas de 5 cts " + moneda);
        centimos = centimos - (moneda * 5);
        moneda = (centimos / 2);
        System.out.println("monedas de 2 cts " + moneda);
        centimos = centimos - (moneda * 2);
        System.out.println("monedas de 1 cts " + moneda);
    
        System.out.println("   fin ejercicio 5");

        System.out.println("CONDICIONALES inicio ejercicio 1");
        //Pedir un número entero al usuario e indicar si es par o impar.
        int numtres=5;
        if (numtres%2==0){
            System.out.println("el número es par");
        }else{
            System.out.println("el número es impar");
        }

        System.out.println("CONDICIONALES    fin ejercicio 1");       

        System.out.println("CONDICIONALES inicio ejercicio 2");
        //Preguntar a un usuario cuántos años tiene y, si es mayor de edad, preguntarle si tiene carné de conducir.
        int edaddos=19;
        if (edaddos>=18){
            System.out.println("¿tienes carnet de conducir?");
        }
         // si ponemos menos de 18 no le preguntamos nada   

        System.out.println("CONDICIONALES    fin ejercicio 2");

        System.out.println("CONDICIONALES inicio ejercicio 3");
        //Pedir al usuario tres valores enteros y mostrar el menor de ellos.
        int aa=3;
        int bb=5;
        int cc=1;
        int menor;
        if (aa<=bb && aa<=cc){
            menor=aa;
        }else if(bb<=aa && bb<=cc){
            menor=bb;
        }else{
            menor=cc;
        }
        System.out.println("el número menor es: "+ menor);


        System.out.println("CONDICIONALES    fin ejercicio 3");

        System.out.println("CONDICIONALES inicio ejercicio 4");
        //Pedir tres números reales al usuario y mostrar el menor o el mayor, según decida.(se va a indicar el menor y el mayor)
        int num1=7;
        int num2=9;
        int num3=5;
        int menos,mas;
        if (num1>=num2 && num1>=num3){
            mas=num1;
            if (num2>=num3){
                menos=num3;
            }else{
                menos=num2;
            }
        }else if (num2>=num1 && num2>=num3){
            mas=num2;
            if (num1>=num3){
                menos=num3;
            }else{
                menos=num1;
            }
        }else{
            mas=num3;
            if (num2>=num1){
                menos=num1;            
            }else {
                menos=num2;
            }    
        }
        System.out.println(("el número menor es : "+ menos +" y el mayor :"+mas));






        System.out.println("CONDICIONALES    fin ejercicio 4");

        System.out.println("CONDICIONALES inicio ejercicio 5");
        /*Escribir un programa que pida un número real de precisión sencilla y haga muestre los siguientes resultados:
        Si el número es menor que 50, mostrará su cuadrado.
        Si el número es igual a 25, 30 ó 75, mostrará su raíz cúbica.
        Si el número vale 10 ó es mayor que 100 ó está en los intervalos [3,8]
        o [77, 90], mostrará el valor dividido por 10.
        Encualquier otro caso, mostrará el número leído.*/
        double real=100;
        int otrocaso=0;
        if (real<50){
            double cuadrado=Math.pow(real,2);
            System.out.println("su cuadrado es : "+cuadrado);
            otrocaso++;
        }
        if (real==25 || real==30 || real==75){
            double raiz=Math.cbrt(real);
            System.out.println("la raíz cúbica es :"+raiz);
            otrocaso++;
        }
        if ((real==10) || (real>100) || (real>=3 && real<=8) || (real>=77 && real<=90)) {
            double division=real/10;
            System.out.println("la división entre 10 es : "+division);
            otrocaso++;
            
        }
        if (otrocaso==0){
            System.err.println("El número introducido es : "+real);
        }


        System.out.println("CONDICIONALES    fin ejercicio 5");

        System.out.println("CONDICIONALES inicio ejercicio ");        
        System.out.println("CONDICIONALES    fin ejercicio ");

    }
}
